rails/app/policies/portal/permission_form_policy.rb

```diff
@@ -30,6 +30,14 @@ def external_report_query?
     manager_or_researcher_or_project_researcher?
   end
 
+  def external_report_query_jwt?
+    manager_or_researcher_or_project_researcher?
+  end
+
+  def external_report_learners_from_jwt?
+    true
```

Comment:
You're right. I had gotten confused on this route before I fully understood that the user's id in the JWT would set the current user. It had seemed like this route could be fully uncredentialed, since it used the user from the JWT to do the part that requires permissions. (Though returning `true` was useless without `authorize`.)

I see now that this would work with `manager_or_researcher_or_project_researcher?`, and it will use the user id in the JWT. So basically if you somehow got ahold of a JWT that was not a manager or researcher, it will fail with a nice 403. If you have a user id that is one of those things but does not have right to the learners in the request, you will get a 200 but no learners.

Reply:
Updated.